Scripts/End/EndPlatformController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EndPlatformController : MonoBehaviour
+{
+
+    [Space, SerializeField]
+    GameObject endPlatform;
+
+    [Space, SerializeField]
+    ParticleSystem smoke;
+
+    [Space, SerializeField]
+    ParticleSystem fire;
+    Collider player;
+
+
+    private void OnTriggerEnter(Collider other)
+    {
+        player = other;
+        PlayerMovement playerMovement = other.GetComponent<PlayerMovement>();
+        playerMovement.setIsGameEnd(true);
+
+        GameObject environment = endPlatform.transform.parent.gameObject;
+        Rigidbody rigidbody = environment.GetComponent<Rigidbody>();
+        rigidbody.constraints = RigidbodyConstraints.FreezePositionZ;
+
+        CheckWater();
+    }
+
+    private void CheckWater()
```

Comment:
Yukarıda sürekli current collider'ı tutmak yerine, methoda inject edilebilir:

```suggestion
    private void CheckWater(Collider playerCollider)
```